src/client/pmix_client.c

```diff
@@ -574,9 +585,21 @@ static void _putfn(int sd, short args, void *cbdata)
         /* shouldn't be possible */
         goto done;
     }
+#if defined(PMIX_ENABLE_DSTORE) && (PMIX_ENABLE_DSTORE == 1)
+    /* TODO: It is not safe to store data on a client side
+     * There is a possibility to get server/client conflict.
+     * Do nothing here misses PMIx_Get/PMIx_Put flow (w/o PMIx_Commit)
+     */
+    /*
+    if (PMIX_SUCCESS != (rc = pmix_dstore_store(ns->nspace, pmix_globals.myid.rank, kv))) {
+        PMIX_ERROR_LOG(rc);
+    }
+    */
+#else
```

Comment:
It is good point to think. Doing this changes I tried to avoid hash->modex usage on client side basing on idea that it is not a sharm way to track both places of data as hash->modex and dstore. So saving data in this place in hash->modex we need to enable a code that reads data from hash->modex at least. I am not a fan of this solution.
Actually I am thinking if we could implement dstore module that operates with hash in addition to esh module. In this case we could have single call for hash or shared memory operates (removing all #if construction).